packages/eslint-plugin/src/rules/no-floating-promises.ts

```diff
@@ -296,12 +327,31 @@ export default createRule<Options, MessageId>({
   },
 });
 
+function isPromiseArray(checker: ts.TypeChecker, node: ts.Node): boolean {
+  const type = checker.getTypeAtLocation(node);
+  for (const ty of tsutils
+    .unionTypeParts(type)
+    .map(t => checker.getApparentType(t))) {
+    if (checker.isArrayType(ty)) {
```

Comment:
I've chosen isArrayType rather than isArrayLikeType because the main real use case here is to catch the result of .map(f), which should always be a native array type.

Looking for advice on whether this was the correct call or ArrayLike would be better.